Formula/swiftlint.rb

```diff
@@ -14,10 +14,15 @@ class Swiftlint < Formula
   end
 
   depends_on xcode: ["11.4", :build]
-  depends_on :macos # Depends on Swift.  May work on Linux once a Swift bottle is available for that OS.
   depends_on xcode: "8.0"
 
+  uses_from_macos "swift"
+
   def install
+    on_linux do
+      ENV["CC"] = Formula["swift"].opt_libexec/"bin/clang"
+      ENV["CXX"] = Formula["swift"].opt_libexec/"bin/clang++"
+    end
```

Comment:
- **Pro**: Avoid extra build-time dependency to `llvm` and potential linkage.
- **Con**: Doesn't use Homebrew's shim.

Reply:
We should teach the shims how to use the Clang that comes with Swift.

It might be nice if `fails_with :gcc` just used any available `clang` from one of the `llvm` or `swift` formulae. CC @Homebrew/brew

Reply:
Good point about the shims - I'll incorporate shim adjustments into the `depends_on :swift` I'm working on just now. I don't think we should be using GCC for anything using Swift.

Consider Swift support as experimental until then.

> It might be nice if `fails_with :gcc` just used any available `clang` from one of the `llvm`

This already happens, just not the one shipped with Swift yet.

Reply:
> We should teach the shims how to use the Clang that comes with Swift.

Agreed.

Reply:
https://github.com/Homebrew/brew/pull/11964, https://github.com/Homebrew/brew/pull/11965